File: src/main/kotlin/magicbook/gtlitecore/common/entity/explosive/EntityLeptonicCharge.java
package magicbook.gtlitecore.common.entity.explosive;

import gregtech.common.entities.EntityGTExplosive;
import magicbook.gtlitecore.common.block.GTLiteMetaBlocks;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.world.World;
import org.jetbrains.annotations.NotNull;

public class EntityLeptonicCharge extends EntityGTExplosive
{

    @SuppressWarnings("unused")
    public EntityLeptonicCharge(World world)
    {
        super(world);
    }

    public EntityLeptonicCharge(World world, double x, double y, double z, EntityLivingBase exploder)
    {
        super(world, x, y, z, exploder);
    }

    @Override
    protected float getStrength()
    {
        return 800.0F;
    }

    @Override
    public boolean dropsAllBlocks()
    {
        return false;
    }

    @Override
    protected int getRange()
    {
        return 1024;
    }

    @NotNull
    @Override
    public IBlockState getExplosiveState()
    {
        return GTLiteMetaBlocks.LEPTONIC_CHARGE.getDefaultState();
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/common/entity/explosive/EntityLeptonicCharge.kt
package gregtechlite.gtlitecore.common.entity.explosive

import gregtech.common.entities.EntityGTExplosive
import gregtechlite.gtlitecore.common.block.GTLiteMetaBlocks
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.EntityLivingBase
import net.minecraft.world.World

class EntityLeptonicCharge(world: World?,
                           x: Double, y: Double, z: Double,
                           exploder: EntityLivingBase?) : EntityGTExplosive(world, x, y, z, exploder)
{
    constructor(world: World?) : this(world, 0.0, 0.0, 0.0, null)

    override fun getStrength() = 800.0f

    override fun dropsAllBlocks() = false

    override fun getRange() = 1024

    override fun getExplosiveState(): IBlockState = GTLiteMetaBlocks.LEPTONIC_CHARGE.defaultState

}
